AmplifyPlugins/Auth/AWSAuthPlugin/Support/Utils/AuthErrorHelper.swift

```diff
@@ -0,0 +1,174 @@
+                                            AWSCognitoAuthError.invalidPassword)
+
+        case .mfaMethodNotFound(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.mfaMethodNotFoundError,
+                                            AWSCognitoAuthError.mfaMethodNotFound)
+
+        case .passwordResetRequired(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.passwordResetRequired,
+                                            AWSCognitoAuthError.passwordResetRequired)
+
+        case .resourceNotFound(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.resourceNotFoundError,
+                                            AWSCognitoAuthError.resourceNotFound)
+
+        case .softwareTokenMFANotFound(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.softwareTokenNotFoundError,
+                                            AWSCognitoAuthError.softwareTokenMFANotEnabled)
+
+        case .tooManyFailedAttempts(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.tooManyFailedError,
+                                            AWSCognitoAuthError.failedAttemptsLimitExceeded)
+
+        case .tooManyRequests(let message),
+             .limitExceeded(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.tooManyRequestError,
+                                            AWSCognitoAuthError.requestLimitExceeded)
+
+        case .errorLoadingPage(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.errorLoadingPageError,
+                                            AWSCognitoAuthError.errorLoadingUI)
+
+        case .deviceNotRemembered(let message):
+            return AmplifyAuthError.service(message,
+                                            AuthPluginErrorConstants.deviceNotRememberedError,
+                                            AWSCognitoAuthError.deviceNotTracked)
+
+        case .invalidState(let message):
+            return AmplifyAuthError.invalidState(message, AuthPluginErrorConstants.invalidStateError)
+
+        case .invalidConfiguration(let message),
+             .cognitoIdentityPoolNotConfigured(let message),
+             .invalidUserPoolConfiguration(let message):
+            return AmplifyAuthError.configuration(message, AuthPluginErrorConstants.configurationError)
+
+        case .notAuthorized(let message):
+            // Not authorized is thrown from server at different scenarios
```

Comment:
This isn't a helpful comment. What is it telling me that i need to know but can't figure out from the clearly-named case?

Reply:
Updated the comments. 
This error is a direct mapping from the service error. And cognito throws this error in many different scenarios like username/pwd is wrong, timeout in signIn etc. That is the reason I initially gave a generic comment.